test/e2e/gang_scheduling/gang_scheduling_test.go

```diff
@@ -725,7 +730,40 @@ func checkCompletedAppStatus(applicationID, state string) {
 	Ω(timeoutErr).NotTo(HaveOccurred())
 }
 
-func checkPlaceholderData(appDaoInfo *dao.ApplicationDAOInfo, tgName string, count, replaced, timeout int) {
+func waitForPlaceholderData(nsQueue string, appID string, tgName string, count, replaced, timedOut, timeout int) error {
+	lastOk := false
+	var lastCount, lastReplaced, lastTimedOut int
+
+	err := wait.PollUntilContextTimeout(context.Background(), 2*time.Second, time.Duration(timeout)*time.Second, false, func(c context.Context) (bool, error) {
+		appDaoInfo, err := restClient.GetAppInfo(configmanager.DefaultPartition, nsQueue, appID)
+		if err != nil {
+			return false, err
+		}
+		lastOk, lastCount, lastReplaced, lastTimedOut = getPlaceholderData(appDaoInfo, tgName)
+		return lastOk && lastCount == count && lastReplaced == replaced && lastTimedOut == timedOut, nil
```

Comment:
To avoid the API call again to assert allocationsCount at caller side, Can we either assert or return allocationsCount?